Ginger/GingerAutoPilot/APIModelLib/ImportParametersOptionalValues.cs

```diff
@@ -0,0 +1,268 @@
+            foreach (XMLDocExtended XDN in NodeList)
+            {
+                AddXMLValueToOptionalValuesPerParameterDict(OptionalValuesPerParameterDict, XDN);
+            }
+        }
+
+        public void GetXMLAllOptionalValuesFromExamplesFile(string fileContent, Dictionary<Tuple<string, string>, List<string>> OptionalValuesPerParameterDict)
+        {
+            XmlDocument XmlDocument = new XmlDocument();
+            XmlDocument.LoadXml(fileContent);
+            XMLDocExtended XDE = new XMLDocExtended(XmlDocument);
+            IEnumerable<XMLDocExtended> NodeList = XDE.GetEndingNodes(false);
+            foreach (XMLDocExtended XDN in NodeList)
+            {
+                AddXMLValueToOptionalValuesPerParameterDict(OptionalValuesPerParameterDict, XDN);
+            }
+        }
+
+        private void AddXMLValueToOptionalValuesPerParameterDict(Dictionary<Tuple<string, string>, List<string>> OptionalValuesPerParameterDict, XMLDocExtended XDN)
+        {
+            Tuple<string, string> tuple = new Tuple<string, string>(XDN.LocalName, XDN.XPath);
+            string Value = XDN.Value;
+
+            if (OptionalValuesPerParameterDict.ContainsKey(tuple))
+            {
+                OptionalValuesPerParameterDict[tuple].Add(Value);
+            }
+            else
+            {
+                OptionalValuesPerParameterDict.Add(tuple, new List<string>() { Value });
+            }
+
+            foreach (XmlAttribute attribute in XDN.Attributes)
+            {
+                Tuple<string, string> attributetuple = new Tuple<string, string>(attribute.LocalName, XDN.XPath);
+                string attributeValue = attribute.Value;
+
+                if (OptionalValuesPerParameterDict.ContainsKey(attributetuple))
+                {
+                    OptionalValuesPerParameterDict[attributetuple].Add(attributeValue);
+                }
+                else
+                {
+                    OptionalValuesPerParameterDict.Add(attributetuple, new List<string> { attributeValue });
+                }
+            }
+        }
+
+        /// <summary>
+        /// Update all parameters optional values according to xml file
+        /// </summary>
+        /// <param name="AAM"></param>
+        /// <param name="OptionalValuesPerParameterDict"></param>
+        public void PopulateXMLOptionalValuesForAPIParameters(ApplicationModelBase AAM, Dictionary<Tuple<string, string>, List<string>> OptionalValuesPerParameterDict)
+        {
+            foreach (AppModelParameter AMP in AAM.AppModelParameters)
+            {
+                string result = Regex.Match(AMP.Path, @"(.)*soapenv:Body\[1\]\/([a-zA-Z]|\d)*:").Value;
+                if (string.IsNullOrEmpty(result))
+                {
+                    result = Regex.Match(AMP.Path, @"(.)*soapenv:Body\[1\]\/").Value;
+                }
+
+                string VAXBXPath = string.Empty;
+                if (!string.IsNullOrEmpty(result))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/Ginger-Automation/Ginger/pullRequest?prid=4464228)